### ISSUE TITLE
Sankey diagrams: Manually set positions applied to the wrong nodes

### ISSUE THREAD

**Issue by johentsch** [plotly/plotly.py#2186]
This is some correct Sankey diagram with some of the positions set manually:

    from numpy import array
    import plotly.graph_objects as go

    go.Figure(go.Sankey({
         'arrangement': 'snap',
         'link': {'source': array([ 0,  0,  0,  1,  1,  1,  1,  1,  1,  1,  1,  1,  2,  2,  2,  2,  2,  2,
                                    2,  2,  2,  3,  3,  3,  4,  5,  5,  5,  5,  5,  6,  8,  8,  8,  9,  9,
                                    9,  9,  9,  9,  9,  9,  9,  9,  9, 10, 10, 10, 10, 10, 10, 10, 10, 10,
                                   11, 11, 11, 11, 12, 12, 12, 13, 13, 13, 14, 14, 14, 14, 14, 14, 14, 14,
                                   14, 14]),
                  'target': array([ 3,  5, 14,  0,  2,  3,  5,  9, 10, 11, 12, 14,  0,  1,  5,  7,  8,  9,
                                   11, 12, 14,  2,  7,  9,  0,  1,  3,  7, 10, 13,  7,  0,  2,  9,  0,  1,
                                    2,  5,  7,  8, 10, 11, 12, 13, 14,  0,  1,  2,  5,  7,  8,  9, 11, 12,
                                    0,  5,  9, 14,  1,  5,  7,  7,  9, 14,  0,  1,  2,  3,  5,  7,  9, 10,
                                   11, 13]),
                  'value': array([  0,  0, 27,  0, 16,  5, 14,  9,  5,  5,  0,  8,  0,  5,  6,  0,  0,  8,  
                                    6,  0, 48, 21,  0,  0,  1, 15,  0, 59,  0,  0,  1,  0,  0,  6,  5, 18,  
                                    0, 20,106,  7,  6,  0,  7, 13, 30,  0,  0,  0,  0,  5,  0,  0,  6,  0,  
                                    0,  0, 43,  0,  6,  0, 10, 55, 15,  0,  5,  7,  0,  0, 20,276,  7,  0,  
                                    0,  6])},
         'node': {'label': [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14],
                  'pad': 20,
                  'x': [0, 0, 0, 0, 0, 0.7, 0, 0.999, 0, 0.7, 0, 0, 0, 0.7, 0.7],
                  'y': [0, 0, 0, 0, 0, 0.001, 0, 0.999, 0, 0.5, 0, 0, 0, 0.001, 0.999]}
     }))

![correct](https://user-images.githubusercontent.com/42718519/74324985-2a5f4880-4d88-11ea-873c-59cd47064de9.png)

The problem starts when I try to filter out the two edges with flow 1 [i.e., (4, 0) and (6, 7)]. In general, we need to substitute node names for our data in order to calculate flow values, so the filtering can be done only after fixing node names. That's why I've tried to exclude the two edges by setting their flow to 0. This, in fact, makes nodes 4 and 6 disappear which leads to the positions (for nodes 5, 7, 9, 13, 14) being applied to the wrong nodes. (Also, there are broken edges now.) Here you go, the same code only with the two 1s changed to 0:

    from numpy import array
    import plotly.graph_objects as go

    go.Figure(go.Sankey({
         'arrangement': 'snap',
         'link': {'source': array([ 0,  0,  0,  1,  1,  1,  1,  1,  1,  1,  1,  1,  2,  2,  2,  2,  2,  2,
                                    2,  2,  2,  3,  3,  3,  4,  5,  5,  5,  5,  5,  6,  8,  8,  8,  9,  9,
                                    9,  9,  9,  9,  9,  9,  9,  9,  9, 10, 10, 10, 10, 10, 10, 10, 10, 10,
                                   11, 11, 11, 11, 12, 12, 12, 13, 13, 13, 14, 14, 14, 14, 14, 14, 14, 14,
                                   14, 14]),
                  'target': array([ 3,  5, 14,  0,  2,  3,  5,  9, 10, 11, 12, 14,  0,  1,  5,  7,  8,  9,
                                   11, 12, 14,  2,  7,  9,  0,  1,  3,  7, 10, 13,  7,  0,  2,  9,  0,  1,
                                    2,  5,  7,  8, 10, 11, 12, 13, 14,  0,  1,  2,  5,  7,  8,  9, 11, 12,
                                    0,  5,  9, 14,  1,  5,  7,  7,  9, 14,  0,  1,  2,  3,  5,  7,  9, 10,
                                   11, 13]),
                  'value': array([  0,  0, 27,  0, 16,  5, 14,  9,  5,  5,  0,  8,  0,  5,  6,  0,  0,  8,  
                                    6,  0, 48, 21,  0,  0,  0, 15,  0, 59,  0,  0,  0,  0,  0,  6,  5, 18,  
                                    0, 20,106,  7,  6,  0,  7, 13, 30,  0,  0,  0,  0,  5,  0,  0,  6,  0,  
                                    0,  0, 43,  0,  6,  0, 10, 55, 15,  0,  5,  7,  0,  0, 20,276,  7,  0,  
                                    0,  6])},
         'node': {'label': [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14],
                  'pad': 20,
                  'x': [0, 0, 0, 0, 0, 0.7, 0, 0.999, 0, 0.7, 0, 0, 0, 0.7, 0.7],
                  'y': [0, 0, 0, 0, 0, 0.001, 0, 0.999, 0, 0.5, 0, 0, 0, 0.001, 0.999]}
     }))

Result:
![sankey](https://user-images.githubusercontent.com/42718519/74324833-eff5ab80-4d87-11ea-8124-f1cd82249343.png)

Expected Behaviour: The plot should be the same as above, only without the nodes 4 and 6.

Plots created with Plotly 4.4.1

**Comment by rubenbfeld-ite** [comment]
Hi laserjeyes,
I think you described the same misbehaviour as I already described here: https://github.com/plotly/plotly.py/issues/2152

The workaround which worked for me was to eliminate the coordinates for the positioning for all nodes with no links or links with value = zero. No real solution but a at least a workaround.

**Comment by gvwilson** [comment]
Hi - we are currently trying to tidy up Plotly's public repositories to help us focus our efforts on things that will help users most. Since this issue has been sitting for several years, I'm going to close it. If it's still a concern, we'd be grateful if you could open a new issue (with a short reproducible example if appropriate) so that we can add it to our backlog. Thanks for your help - @gvwilson
